wolfcrypt/src/port/Renesas/renesas_tsip_rsa.c

```diff
@@ -0,0 +1,300 @@
+        }
+        
+        tsip_hw_unlock();
+    }
+
+    return 0;
+}
+
+
+/* Perform Rsa verify by TSIP
+ * Assumes to be called by Crypt Callback
+ * 
+ * in     Buffer to hold plaintext
+ * inLen  Length of plaintext in bytes
+ * out    Buffer to hold generated signature
+ * outLen Length of signature in bytes
+ * key    rsa key object
+ * ctx    The callback context
+ * return FSP_SUCCESS(0) on Success, otherwise negative value
+ */
+ 
+WOLFSSL_LOCAL int wc_tsip_RsaVerifyPkcs(wc_CryptoInfo* info, TsipUserCtx* tuc)
+{
+    int ret = 0;
+    e_tsip_err_t    err = TSIP_SUCCESS;
+    tsip_rsa_byte_data_t hashData, sigData;
+
+    uint8_t  tsip_hash_type;
+    
+    
+    /* sanity check */
+    if (info == NULL || tuc == NULL){
+        return BAD_FUNC_ARG;
+    }
+    
+    if (ret == 0) {
+       if (tuc->sing_hash_type == md5_mac)
+           tsip_hash_type = R_TSIP_RSA_HASH_MD5;
+       else if (tuc->sing_hash_type == sha_mac)
+           tsip_hash_type = R_TSIP_RSA_HASH_SHA1;
+       else if (tuc->sing_hash_type == sha256_mac)
+           tsip_hash_type = R_TSIP_RSA_HASH_SHA256;
+       else
+           ret = CRYPTOCB_UNAVAILABLE;
+    }
+    
+    switch (tuc->wrappedKeyType) {
+        case TSIP_KEY_TYPE_RSA1024:
+            if (tuc->keyflgs_crypt.bits.rsapub1024_key_set != 1)
+            {
+                ret = tsipImportPublicKey(tuc, tuc->wrappedKeyType);
+
+                WOLFSSL_MSG("tsip rsa private key 1024 not set");
+                if (ret != 0) 
+                    ret = CRYPTOCB_UNAVAILABLE;
+
+            }
+            break;
+        case TSIP_KEY_TYPE_RSA2048:
+            if (tuc->keyflgs_crypt.bits.rsapub2048_key_set != 1)
+            {
+                ret = tsipImportPublicKey(tuc, tuc->wrappedKeyType);
+
+                WOLFSSL_MSG("tsip rsa private key 1024 not set");
+                if (ret != 0) 
+                    ret = CRYPTOCB_UNAVAILABLE;
+            }
+            break;
+        default:
+            WOLFSSL_MSG("wrapped private key is not supported");
+            ret = CRYPTOCB_UNAVAILABLE;
+            break;
+    }
+
+    if (ret == 0) {
+        hashData.pdata = (uint8_t*)info->pk.rsa.in;
+        hashData.data_length = info->pk.rsa.inLen;
+        hashData.data_type =
+            tuc->keyflgs_crypt.bits.message_type;/* message 0, hash 1 */
+
+        sigData.pdata = (uint8_t*)info->pk.rsa.out;
+        sigData.data_length = info->pk.rsa.outLen;
+
+        if ((ret = tsip_hw_lock()) == 0) {
+            switch (tuc->wrappedKeyType) {
+                case TSIP_KEY_TYPE_RSA1024:
+                    err = R_TSIP_RsassaPkcs1024SignatureVerification(&sigData,
+                            &hashData,
+                            (tsip_rsa1024_public_key_index_t *)
+                            tuc->rsa1024pub_keyIdx,
+                            tsip_hash_type);
+                    if (err == TSIP_ERR_AUTHENTICATION) {
+                            ret = SIG_VERIFY_E;
```

Comment:
This line needs to move back one indent (4 spaces).

Reply:
Thanks, Fixed